wordpress-mobile/WordPress-Android
# [Nullability Annotations to Java Classes] Use Updated and Null Proof `Theme` Payload Classes (`safe`)

Parent [FluxC#2798](https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/2798)
Accompanying [FluxC#2893](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2893)

This PR accompanying the above FluxC PR, and uses the updated and null-proof `Theme` payload classes.

FYI: This change is `safe`, meaning that there shouldn't be any (major) compile-time changes associated with this change.

-----

@antonis I added you as the main reviewer, not so randomly, due to the fact that you're also reviewing [FluxC#2893](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2893), and I wanted you from the Jetpack/WordPress mobile team again to be aware of and sign-off on that change. Feel free to merge this PR directly yourself if you deem so.

-----

FluxC Update List:

1. [Update fluxc version to pr hash (2893)](https://github.com/wordpress-mobile/WordPress-Android/commit/6abaf1354e69b9428475ddfcb4bea38f33af42b9)
2. [Update fluxc version to pr hash (2893)](https://github.com/wordpress-mobile/WordPress-Android/pull/19529/commits/a2d486d1c27ecdff555d21da452fb9d5c81b7b1f)
3. [Update fluxc version to trunk hash (2893)](https://github.com/wordpress-mobile/WordPress-Android/pull/19529/commits/7f546860c2b4c3799a33e8e1d358c4d7e02eeb22)

Nullability Checks List:

1. [Un-guard usages of non-null on starter designs fetched designs](https://github.com/wordpress-mobile/WordPress-Android/commit/5381456f806e35ef46717e1ffc79e2a9c908b02a)

Test List:

1. ~~[Resolve blaze campaign model parameter target urn related warnings](https://github.com/wordpress-mobile/WordPress-Android/commit/9c21ef2cc189db42e5b96dc9ddecef2e40ad7101) (`unrelated`)~~

-----

## To Test (`REST`):

- Smoke test any theme related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected. For a couple of examples, you can expand and follow the inner and more explicit test steps within:

## To Test (`XMLRPC`):

N/A

<details>
    <summary>1. [JP/WP] Themes Screen [ThemeBrowserActivity.java + ThemeBrowserFragment.kt]</summary>

ℹ️ This test applies to both, the `Jetpack` and `WordPress` apps.

- Go to `Themes` screen, verify it is shown and functioning as expected.
- For example try scrolling up-and-down and searching for a theme.
- Check your current theme and tap on the `Customize`, `Details` and `Support` buttons. Verify that everything works as expected.
- From the themes list, find your current theme on top, click on its menu items and again tap on the `Customize`, `Details` and `Support` buttons. Verify that everything works as expected.
- From the themes list, find another theme, other than your current one, click on its menu items and again tap on the `Details` and `Support` buttons. Verify that everything works as expected.
- Then, tap on the `View` and `Try & Customize` buttons. Verify that everything works as expected.
- Finally, tap on `Activate` button. Verify that the selected theme is activated and that everything works as expected.

</details>

<details>
    <summary>2. [JP] Site Creation Screens [SiteCreationActivity.kt + HomePagePickerFragment.kt + DesignPreviewFragment.kt]</summary>

ℹ️ This test applies to the `Jetpack` app only.

- From `Choose Site`, tap on the `+` button on top and via the `Create WordPress.com site`, go to `Site Creation` screen, verify it is shown and functioning as expected.
- For example, while on the `What's your website about?` screen, select a topic from the list of topics (ie. `Food`).
- Then, while on the `Choose a theme` screen, select a theme from the list of themes (per category, ie `About`).
- Finally, verify that the `Preview` screen is shown and functioning as expected.

</details>

-----

# Merge instructions:

1. [x] Wait for the accompanying [FluxC#2893](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2893) PR to be merged.
4. [x] Update `wordPressFluxCVersion` to point to the `trunk` hash that includes the above solution.
5. [x] Remove the `[Status] Not Ready for Merge` label.
6. [x] Merge this PR.

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

4. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR submission checklist:


-----

## UI Changes testing checklist:
